build-system/app-index/components/proxy-form.js

```diff
@@ -0,0 +1,75 @@
+/**
+ * Copyright 2018 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* eslint-disable no-unused-vars */
+
+import {Component, h, render} from 'preact';
+
+class ProxyForm extends Component {
+
+  constructor() {
+    super();
+    this.setState({
+      proxyInput: '',
+    });
+  }
+
+  handleProxyInputChange(event) {
+    this.setState({
+      ...this.state,
+      proxyInput: event.target.value,
+    });
+  }
+
+  handleSubmit(event) {
```

Comment:
Can use the arrow syntax when defining this method and autobind this to avoid doing so in the render loop.

```javascript
handleSubmit = event => {
  event.preventDefault();
  window.location = `/proxy/s/${this.state.proxyInput.replace(/^http(s?):\/\//i, '')}`;
}
```